gammapy/cube/fit.py

```diff
@@ -28,8 +35,14 @@ class MapDataset:
         Energy dispersion
     background_model: `~gammapy.cube.models.BackgroundModel` or `~gammapy.cube.models.BackgroundModel`
         Background models to use for the fit.
-    likelihood : {"cash"}
+    likelihood : {"cash", "cstat"}
 	    Likelihood function to use for the fit.
+	evaluation_mode : {"local", "global"}
+        Model evaluation mode. The "cutout" mode evaluates the model components on smaller grids
```

Comment:
I don't find the distinction between local and global optimization algorithms very clear. Is it an issue of algorithms or available IRF granularity? 
In a sense you could still have 2 PSF kernels for two distinct sources and still evaluate/convolve on the full grid.

Reply:
Also keep `local`rather than `cutout`

Reply:
The problem that arises here is the caching of the coordinate grids. The way it is implemented now, the `MapEvaluator` computes the coordinate grids and caches it for model evaluation. If each of the `MapEvaluator` for every model component caches the full coordinate grid this quickly becomes inefficient. The solution here is probably to compute the global coordinate grid on the `MapDataset` and make `cutouts` (as views) into the original array which are the passed to the `MapEvaluator`. I'll try to change the implementation.

Reply:
Or we implement the caching of the `.get_coord()` method on the `MapGeom` object...

Reply:
OK. But if you are to perform oversampling the initial coordinate arrays becomes irrelevant, right?
I agree that making `MapGeom.get_coord()` a `@lazyproperty` seems like a good idea.

Reply:
Yes, oversampling for the global mode will not be supported. However I changed the implementation of the global mode to set a PSF / Edisp per model component, which is updated as well if the model is too far from it's initial position. I've thought about it again and as we only cache the spatial coordinates (and not the full 3D coordinate array), it's probably OK to keep the full 2D coordinate array per model component. It should not be a problem as long as the number of sources is ~10-20.